flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -392,6 +392,7 @@ protected YarnClient getYarnClient() {
 	@Override
 	public YarnClusterClient retrieve(String applicationID) {
 
+		final YarnClient yarnClient = getYarnClient();
 		try {
 			// check if required Hadoop environment variables are set. If not, warn user
 			if (System.getenv("HADOOP_CONF_DIR") == null &&
```

Comment:
does `getYarnClient()` fail if this condition is fulfilled? If so we should put this if block before the call to `getYarnClient()`.